huxley/api/serializers/delegate.py

```diff
@@ -25,6 +31,32 @@ class Meta:
                   'session_three',
                   'session_four', )
 
+    def update(self, instance, validated_data):
+        if ('assignment' in validated_data and
+                validated_data['assignment'] != None and
+                not instance.assignment):
+
+            names = instance.name.split(' ')
+            username = names[0] + '_' + str(instance.id)
+            password = BaseUserManager().make_random_password(10)
+            user = User.objects.create(delegate=instance, username=username,\
+                                       user_type=User.TYPE_DELEGATE,\
+                                       last_login=datetime.now(),
+                                       first_name=names[0], last_name=names[-1],
+                                       email=instance.email)
+            user.set_password(password)
+            user.save()
+            send_mail('A new account has been created for {0}!\n'.format(instance.name),
+                      'Username: {0}\n'.format(username) \
+                      + 'Password: {0}\n'.format(password) \
```

Comment:
Yikes, plaintext passwords. But, since delegates aren't creating their own accounts, I don't see an immediate way around this, other than forcing them to change their password the first time they log in.

Reply:
I agree it isn't great, but the accounts themselves contain nothing sensitive and we need an efficient way to distribute the passwords.